common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -257,6 +258,159 @@ public int sslShuffleChunkSize() {
       conf.get("spark.network.ssl.maxEncryptedBlockSize", "64k")));
   }
 
+  /**
+   * Whether Secure (SSL/TLS) RPC (including Block Transfer Service) is enabled
+   */
+  public boolean sslRpcEnabled() {
+    return conf.getBoolean("spark.ssl.rpc.enabled", false);
+  }
```

Comment:
Typically, there are two cases -
a) Use of signed certificates, which can be validated, 
b) Use of self-signed certificates.

Given the configs documented here, how are we planning to support these ?

Reply:
We support both of these. 

If you like to use the JDK SSL provider, we require a `keyStore` argument which takes in the key being used for the server, and the `trustStore` is used to verify the peer we talk to.

If the openSSL provider is used, we require the `privateKey` and `certChain` arguments which have similar purposes.

You can look at the `SSLFactory` in the reference PR https://github.com/apache/spark/pull/42685 (I was going to put it up after this PR and another one) to see how it's used.

The tests in the reference PR https://github.com/apache/spark/pull/42685 use self signed certificates and work fine. 

Does that answer your question? Happy to clarify as needed

Reply:
Ok, so we are relying on the fact the if truststore is not specified, it implies self signed cert as we are going to bypass cert check.
That sounds fine to me given this is different from 'typical' usage where client is accepting server's cert and does not know if it will be trusted, or should bypass trust (if cert is trusted, use that - but if it no, should it still be trusted).

Reply:
Actually, looking at it again, I want to make sure I am not missing something here.

Currently, `sslRpcEnabledAndKeysAreValid` will return `false` when key store is specified, but trust store is not - and so ssl factory will be `null`.

Within ssl factory, we do have code to handle the case of trust store being null resulting in accepting all server certs - but this will never get triggered since factory is null.

What am I missing here ?